optuna/samplers/_tpe/sampler.py

```diff
@@ -893,3 +891,41 @@ def _calculate_weights_below_for_multi_objective(
     weights_below_all[feasible_mask] = weights_below
     weights_below_all = weights_below_all[~np.isnan(cvals)]
     return weights_below_all
+
+
+def _clamp_inf_and_calc_reference_point(lvals: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+    """Clamp inf values in lvals and calculate the reference point.
+
+    This function satisfies the following axioms:
+    1. Clamped lvals and the calculated reference point only contain finite values.
+    2. Clamping preserves the original order of lvals.
+    3. Clamped lvals are always smaller than or equal to the calculated reference point.
+    """
+    worst_point = np.array(
+        [
+            np.max(lvals[:, i][lvals[:, i] != np.inf], initial=-np.inf)
+            for i in range(lvals.shape[1])
+        ]
+    )
+    worst_point[worst_point == -np.inf] = 0.0
+    posinf_replacement = np.maximum(
+        np.maximum(1.1 * worst_point, 0.9 * worst_point), worst_point + EPS
+    )
```

Comment:
A nit.
```suggestion
    posinf_replacement = np.maximum.reduce([1.1 * worst_point, 0.9 * worst_point, worst_point + EPS])
```

Reply:
I want a code comment here as well. I guess the first element is for the positive `worst_point` values, and the second one is for negative ones, and third one is for zeros.

Reply:
Thanks, fixed.